app/controllers/users_controller.rb

```diff
@@ -290,8 +241,23 @@ def auth_success
         else
           failed_login t("sessions.new.auth failure")
         end
+      elsif user.nil? && user = User.find_by(:email => email)
```

Comment:
I don't think we need `user.nil?` here as we would have taken the previous branch if user was set.